### PR TITLE
feat(migration): Dispatcher config conversion for AEMaaCS — Branch E 

### PR DESCRIPTION
**Pull request by Himanich**
## Description

Adds Branch E — Dispatcher Conversion to the AEM Cloud Service migration skill: converts AMS / on-premise Apache + Dispatcher configurations to AEM as a Cloud Service. It is agent-driven and wraps Adobe's maintained @adobe/aem-cs-source-migration-dispatcher-converter as the deterministic engine — adding the detection, verification, judgment, and reporting layers the raw tool lacks — and hands off to the existing dispatcher skill rather than duplicating it.

## How Has This Been Tested?

Unit suites + an env-gated real-config acceptance test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by rombert — COMMENTED** [review]
I think @abhishekgarg18 is best suited to review here. You should also see why PR checks failed.

File: plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+function extractBraceBody(txt, openIdx) {
+  let depth = 0, i = openIdx, start = openIdx + 1;
+  for (; i < txt.length; i++) {
+    if (txt[i] === '{') depth++;
+    else if (txt[i] === '}') { depth--; if (depth === 0) return txt.slice(start, i); }
+  }
+  return txt.slice(start);
+}
+
+// Count filter/ACL rules across BOTH sources they can live in:
```
Comment (abhishekgarg18) [inline plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js]: The ACL gate is still path-based and misses valid custom `$include` layouts. Please resolve included filter files recursively instead of inferring safety from filenames.

File: plugins/aem/cloud-service/skills/migration/scripts/dispatcher-verify.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  // 2. Rewrite reconciliation (warning-level; rewrites may legitimately move to CDN). Count via
+  //    the shared counter, EXCLUDING Adobe-managed default_* immutables (default_rewrite.rules) so
+  //    this warning agrees with the coverage report's custom-only rewrite figure.
+  const outRw = countRewrites(outputSrcDir, { excludeDefault: true });
+  if ((baseline.rewrite || 0) > outRw) {
+    warnings.push(`Rewrite/redirect count dropped ${baseline.rewrite} → ${outRw} — expected when the source uses .tmpl templates (inflated baseline) or when redirects move to the CDN edge; reconcile against the tool's conversion-report.md rather than treating the delta as loss.`);
+  }
+
+  // 3. Artifact health — oversized vhost (mega-inlined).
+  for (const v of readTextFiles(path.join(outputSrcDir, 'conf.d'), n => n.endsWith('.vhost'))) {
```
Comment (abhishekgarg18) [inline plugins/aem/cloud-service/skills/migration/scripts/dispatcher-verify.js]: This vhost-health check only inspects `conf.d/*.vhost` and ignores the standard `conf.vhost.d/*.conf` layout. Please validate both layouts to avoid missing oversized inline vhosts.

**Review by abhishekgarg18 — CHANGES_REQUESTED** [review]
Hey @Himanich,

## Issues

### Critical (Must Fix)
- plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js:128-167
  - The ACL gate still relies on file naming conventions (`filters.any`, `*_filters.any`, `filters/`) instead of resolving actual `$include` chains. A valid custom include layout can therefore be counted as zero and the security gate will silently pass.
  - This is a security regression risk for dispatcher conversion. A conversion that drops ACLs can look "safe" even though the effective filter policy has been emptied.
  - Fix by resolving included `.any` files recursively and counting the actual ACL entries from the resolved config graph, not just the path names.

- plugins/aem/cloud-service/skills/migration/scripts/dispatcher-verify.js:34-43
  - The "mega-inlined vhost" health check only scans `conf.d/*.vhost`, so the standard `conf.vhost.d/*.conf` layout is skipped.
  - In standard dispatcher trees, that means the validation can miss oversized inline vhosts, which is exactly the type of config that becomes hard to maintain and reason about.
  - Fix by scanning both `conf.d` and `conf.vhost.d`, and include standard `.conf` files in the same check.

### Important (Should Fix)
- plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js:154-166
  - The custom include path detection is brittle. Any include file that is legitimate but not named like `filters.any` or located in a folder named `filters` will be ignored.
  - That makes the verification incomplete for real-world migration layouts and weakens trust in the generated conversion report.

File: plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js (hunk 1 above)
Comment (Himanich) [reply plugins/aem/cloud-service/skills/migration/scripts/dispatcher-inventory.js]: Confirmed and fixed in ee183cc. `countFilterRules` now parses `$include "path"` out of each `/filter` body and resolves it relative to the including file — recursively, so a multi-hop chain is fully walked — regardless of naming. Verified against your exact concern: a custom include like `$include "../acl-policies/prod-rules.any"` (matching no naming convention) now counts correctly (previously 0).

The naming-heuristic scan (`filters.any`/`*_filters.any`/a `filters/` dir) is kept only as a residual safety net for files never reached via a resolvable include — files already counted via a real $include are skipped there to avoid double-counting.

Also added: an $include target that can't be read from disk (absolute container path, a glob, a missing file) is no longer silently treated as "0 rules = safe" — it's collected into a new `ruleCounts.filterIncludesUnresolved` list, and `verifyOutput` now surfaces a warning when either side has one.

Verified end-to-end against a real customer config (86 filter rules) — the gate still correctly fires `filter-acl-loss` on an emptied output, and old vs. new code agree on that config's count (86 == 86), confirming no double-counting was introduced. 5 new regression tests added; 99/99 pass.

**Comment by Himanich** [comment]
Thanks for the review, @abhishekgarg18 — both Critical items are addressed in ee183cc (replied inline on each thread with specifics).

**Important (should-fix): "brittle custom include path detection"** — this is resolved as a byproduct of the Critical #1 fix. The primary counting path no longer depends on file naming at all; it parses and resolves the actual `$include "path"` out of each `/filter` body. Naming (`filters.any`/`*_filters.any`/a `filters/` dir) is now only used as a residual safety net for a file that exists on disk but was never reached via any resolvable include — an edge case, not the primary mechanism.

**Verification on this push:**
- 99/99 unit tests pass (5 new regression tests: a custom non-conventional include path, a multi-hop include chain, an unresolvable include being reported rather than silently zeroed, the same on the output side surfacing as a `verifyOutput` warning, and a `.conf`-suffixed mega-inlined vhost now being flagged).
- Ran the real Adobe converter fresh against a real customer config end-to-end (not just unit fixtures): 86 real filter rules in the baseline, gate still correctly fires `filter-acl-loss` on the emptied output, and old vs. new code agree exactly on that config's count (86 == 86) — confirming the fix doesn't introduce double-counting.

Ready for re-review.

**Review by abhishekgarg18 — APPROVED** [review]
lgtm!!
